Add docs for run dapr without cli. (#1310)
dapr/docs: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy-without-cli.md

---

> type: docs
> title: "Deploy Dapr at a Kubernetes cluster"
> linkTitle: "Deploy Dapr without CLI"
> weight: 21000
> description: "Follow these steps to deploy Dapr at Kubernetes without cluster permissions."
> aliases:
>     - /getting-started/install-dapr-kubernetes/

Reviewer: ```suggestion
type: docs
title: "Deploy Dapr on a Kubernetes cluster without CLI"
linkTitle: "Deploy Dapr without CLI"
weight: 21000
description: "Follow these steps to deploy Dapr at Kubernetes without cluster permissions and the CLI"
```

---

> This part describes the process of installation Dapr without CLI or Helm when you have no cluster permissions. Such situation may arise when you have a project in OKD or namespace in Kubernetes. In this case you have no cluster permission and can't deploy the operator. You can't also address cluster admin to install Dapr operator because you are the only one who needs it.

Reviewer: ```suggestion
This page describes the process of installation Dapr without CLI or Helm when you have no cluster permissions. Such situation may arise when you have a project in OKD or a namespace in Kubernetes. In this case you have no cluster permission and can't deploy the operator. You can't also address cluster admin to install Dapr operator because you are the only one who needs it.
```

Reviewer 2: Just to be clear, this example only deploys a local development environment to the kubernetes cluster, so this should be  explicitly mentioned in the title. This is not about setting up a production environment which should use Helm or the CLI. This is not clear in the overview and it would be good to add.

Author: I agree with you about example. But you could deploy three instances of placement and dashboard. It will be similar to production environment

This approach can also be used with other orchestrators, for example in Docker Swarm.

## Prerequisites

- Install [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
- Kubernetes cluster (see below if needed)

### Create cluster

You can install Dapr at any Kubernetes cluster. Here are some helpful links:

- [Setup Minikube Cluster]({{< ref setup-minikube.md >}})
- [Setup Azure Kubernetes Service Cluster]({{< ref setup-aks.md >}})
- [Setup Google Cloud Kubernetes Engine](https://cloud.google.com/kubernetes-engine/docs/quickstart)
- [Setup Amazon Elastic Kubernetes Service](https://docs.aws.amazon.com/eks/latest/userguide/getting-started.html)

## Install without Dapr CLI

You can install Dapr to a Kubernetes cluster or other orchectrator without Dapr CLI.

### Install Dapr

> Dapr installation is done with ordinary Kubernes deployment. 

Reviewer 2: Typo Kubernes should be Kubernetes. 

I suggest this sentence is reworded to  "Dapr is installed as a Kubernetes deployment."

        - name: DAPR_HTTP_PORT
          value: "3500"
        - name: DAPR_GRPC_PORT
          value: "50001"
        image: application:latest
        name: application
        ports:
        - containerPort: 3000
        resources:
          limits:
            cpu: 200m
            memory: 200Mi
          requests:
            cpu: 200m
            memory: 200Mi
      enableServiceLinks: false
      imagePullSecrets:
      - name: zyfra
      volumes:
      - configMap:
          defaultMode: 420
          name: application-components-statestore
        name: application-components-statestore

apiVersion: v1
kind: Service
metadata:
  annotations:
    description: Exposes and load balances the application pods
  labels:
    app: application
    name: application
  name: application-dapr
spec:
  ports:
  - name: dapr-http
    port: 80
    protocol: TCP
    targetPort: 3500
  - name: dapr-grpc
    port: 50001
    protocol: TCP
    targetPort: 50001
  - name: dapr-internal
    port: 50002
    protocol: TCP
    targetPort: 50002
  selector:
    name: application
```
The next example shows how to apply it to Kubernetes.

```bash
kubectl apply -f application.yml
```

> As you can see, to run application we use mode named **dns**. In this mode to locate the Dapr application we use services with suffix `-dapr`.

Reviewer 2: Suggest reword this to
"To run the application use the **dns** mode . In this mode to locate the Dapr application, use services with suffix `-dapr`.
